velox/exec/HashAggregation.cpp

```diff
@@ -485,6 +514,15 @@ void HashAggregation::reclaim(
           << ", reservation: " << succinctBytes(pool()->reservedBytes());
       return;
     }
+    if (isDistinct_) {
```

Comment:
If we have seen all input and haven't started spilling and isDistinct_ is true, we can just drop the hash table, no?

Reply:
Yeah, we could do. There is a small time widow and most of time, the operator will just close soon. Add to reset table here.